src/app/user/user-login/user-login.component.html

```diff
@@ -0,0 +1,39 @@
+<mat-card>
+<form (submit)="onSubmit(newForm)" #newForm="ngForm">
+	  <!-- username -->
+      <mat-form-field appearance = "legacy" color = "accent">
+          <input 
+            matInput 
+            placeholder = "Username"
+            name = "username"
+            ngModel
+            required
+            minlength = {{minAttributeLength}}
+            #name="ngModel">
+          <mat-hint> At least {{minAttributeLength}} characters </mat-hint>
+          <mat-error *ngIf = "name.invalid"> Username is should be at least {{minAttributeLength}} characters long </mat-error>
+      </mat-form-field>
+
+      <!-- password -->
+      <mat-form-field appearance = "legacy" color = "accent">
+          <input 
+            matInput 
+            placeholder = "Password"
+            name = "password"
+            type = "password"
+            ngModel
+            required
+            minlength = {{minAttributeLength}}
+            #name="ngModel">
+          <mat-hint> At least {{minAttributeLength}} characters </mat-hint>
+          <mat-error *ngIf = "name.invalid"> Password is should be at least {{minAttributeLength}} characters long </mat-error>
```

Comment:
From where did u copy that? (There it is wrong too)
The template name (#) should be #password not #name for password field.
So change lines 27 and 29(BTW did u check that? It should has behavs strangely)

Reply:
Copied it from alex, not really sure about what it means.
I'll change that.

Reply:
Almost there... change this to password .
From phone its not easy to do this for u